### ISSUE TITLE
Required emacs version may need an update

### ISSUE THREAD

**Issue by formido** [skeeto/.emacs.d#4]
I'm running 24.3 and got this error:

```
Debugger entered--Lisp error: (error "Package `emacs-24.4' is unavailable")
  signal(error ("Package `emacs-24.4' is unavailable"))
  error("Package `%s-%s' is unavailable" "emacs" "24.4")
  package-compute-transaction((magit) ((emacs (24 4)) (async (20150909 2257)) (dash (20151021 113)) (with-editor (20160408 201)) (git-commit (20160425 430)) (magit-popup (20160512 328))))
  package-install(magit)
  use-package-ensure-elpa(magit)
  (progn (require (quote package)) (use-package-ensure-elpa (quote magit)))
  (progn (progn (require (quote package)) (use-package-ensure-elpa (quote magit))) (if (fboundp (quote magit-status)) nil (autoload (function magit-status) "magit" nil t)) (let ((body (function (lambda nil (setq magit-last-seen-setup-instructions "2.1.0"))))) (if debug-on-error (funcall body) (condition-case err (funcall body) (error (i$
  eval-buffer(#<buffer  *load*> nil "/home/ubuntu/emacs-configs/skeeto/.emacs.d/init.el" nil t)  ; Reading at buffer position 7216
  load-with-code-conversion("/home/ubuntu/emacs-configs/skeeto/.emacs.d/init.el" "/home/ubuntu/emacs-configs/skeeto/.emacs.d/init.el" t t)
  load("/home/ubuntu/emacs-configs/skeeto/.emacs.d/init" t t)
  #[0 "^H\205\262^@     \306=\203^Q^@\307^H\310Q\202;^@ \311=\204^^^@\307^H\312Q\202;^@\313\307\314\315#\203*^@\316\202;^@\313\307\314\317#\203:^@\320\nB^R\321\202;^@\316\322^S\323^A\322\211#\210^K\322=\203a^@\324\325\326\307^H\327Q!\"\323^A\322\211#\210^K\322=\203`^@^A^S\210^K\203\243^@\330^K!\331\232\203\243^@\332^K!\211\333P\334^A!\2$
  command-line()
  normal-top-level()
```

**Comment by skeeto** [comment]
Nothing in the config itself requires 24.4, but Magit currently requires
Emacs 24.4 at a minimum. Debian Jessie (currently stable) ships 24.4,
making it the oldest version I care about for my own config. Since I
don't use 24.3, I don't care if my config supports it or not.

However, for the packages I maintain, those intended to be used by
others in addition to myself, the oldest version worth supporting is
24.3 (March 2013), and I try to honor that. 24.3 came with enough
changes that supporting anything earlier is a lot of trouble.

**Comment by formido** [comment]
Cool. I was trying to run your config to figure out where `expose` comes from, because when I copy/pasted your `skewer-mode` bits, I got an error on that. The emacs24 package for Ubuntu 14 only goes up to 24.3, so I swapped out my dev instance for Ubuntu 16 and everything's great.

Honestly, your config is better than the one I've haphazardly pieced together over the years, so I may just try running it for awhile anyway.
